hypertension/dc-7101-algorithms/py-root/test/bp_sufficiency_test.py

```diff
@@ -1,5 +1,158 @@
 import pytest
-from lib.algorithms.bp_sufficiency import sufficient_to_autopopulate
+from lib.algorithms.bp_sufficiency import sufficient_to_autopopulate, bp_readings_meet_date_specs, calculate_predominant_ratings
+
+@pytest.mark.parametrize(
+    "date_of_claim, bp_readings, result",
+    [
+        # Valid reading
+        (
+            "2021-11-09",
+            [
+                {
+                    "diastolic": 115,
+                    "systolic": 180,
+                    "date": "2021-10-10"
+                },
+                {
+                    "diastolic": 110,
+                    "systolic": 200,
+                    "date": "2021-09-01"
+                }
+            ],
+            True
+        ),
+        # Reading within 30 days, no reading within 180 days (181 days)
+        (
+            "2021-11-09",
+            [
+                {
+                    "diastolic": 115,
+                    "systolic": 180,
+                    "date": "2021-10-10"
+                },
+                {
+                    "diastolic": 110,
+                    "systolic": 200,
+                    "date": "2021-05-12"
+                }
+            ],
+            False
+        ),
+        # Reading within 180 days, no reading within 30 days (reading has 31 days)
+        (
+            "2021-11-09",
+            [
+                {
+                    "diastolic": 115,
+                    "systolic": 180,
+                    "date": "2021-10-09"
+                },
+                {
+                    "diastolic": 110,
+                    "systolic": 200,
+                    "date": "2021-09-01"
+                }
+            ],
+            False
+        ),
+        # 1 reading
+        (
+            "2021-11-09",
+            [
+                {
+                    "diastolic": 115,
+                    "systolic": 180,
+                    "date": "2021-10-10"
+                }
+            ],
+            False
+        ),
+        # 0 readings
+        (
+            "2021-11-09",
+            [],
+            False
+        )
+    ],
+)
+def test_bp_readings_meet_date_specs(date_of_claim, bp_readings, result):
+    """
+    Test function that determines if the blood pressure readings contain a readings that are within 1 month and 6 months of the date of claim
+
+    :param date_of_claim: string representation of the date of claim
+    :type date_of_claim: string
+    :param bp_readings: list of blood pressure readings
+    :type bp_readings: list
+    :param result: boolean describing whether or not the blood pressure readings meet the specifications
+    :type result: bool
+    """
+    assert bp_readings_meet_date_specs(date_of_claim, bp_readings) == result
+
+
+@pytest.mark.parametrize(
```

Comment:
new test for the refactored algorithm function

Reply:
Got it.

(Resolved)